frontend/src/firebase/firebaseInteractor.ts

```diff
@@ -114,31 +114,65 @@ export default class FirebaseInteractor {
-    let assessment = assessmentDocs[0];
-    let { id, currentIndex } = assessment.data();
-    let wordPath = `assessments/${assessment.id}/words`;
-    let wordRef = await this.db.collection(wordPath).get();
-    let wordDocs = wordRef.docs;
-
-    let words: Word[] = [];
-    for (let wordRef of wordDocs) {
+    let assessment = assessmentRef.data();
+    if (assessment == null) {
+      throw new Error(`Assessment with id ${firebaseId} does not exist`)
+    }
+    let { id, currentIndex, words } = assessment;
+
+    let actualWords: Word[] = [];
+    for (let wordString of words) {
+      let wordRef = await this.db.collection("words").doc(wordString).get();
       let word = wordRef.data();
-      words.push({
+      if (word == null) {
+        throw new Error(`There is no word with the id ${wordString}`);
+      }
+      actualWords.push({
         value: word.value,
         correctImage: word.correctImage,
         incorrectImages: word.incorrectImages,
         id: wordRef.id,
         createdAt: word.dateCreated.toDate(),
       });
     }
-    words.sort(
+    actualWords.sort(
       (word1, word2) => word1.createdAt.getTime() - word2.createdAt.getTime()
     );
-    return { id, currentIndex, words };
+    return { id, currentIndex, words: actualWords, firebaseId: assessment.id };
+  }
+
+  async updateAssessment(id: string, responses: AssessmentResult[]) {
+    await Promise.all(responses.map(async (response) => {
+      return await this.db.collection("assessments").doc(id).collection("results").doc(response.word).set({correct: response.correct});
+    }))
+  }
+
+  async createInterventionSet(id: string, responses: AssessmentResult[]) {
+    await Promise.all(responses.map(async (response) => {
+
+      let word = (await this.db.collection("words").doc(response.word));
+      let wordData = (await word.get()).data();
+      if (wordData == null) {
+        throw Error("This word does not exist");
+      }
+      let wordValue = wordData.value;
+      let correctURL = wordData.correctImage;
+      let incorrectURL = wordData.incorrectImages[Math.floor(Math.random() * wordData.incorrectImages.length)];
+      let incorrectURL2 = wordData.incorrectImages[Math.floor(Math.random() * wordData.incorrectImages.length)];
+
+      let activity3Answer = Math.random() > .5 ? true : false;
+      let activity3URL = activity3Answer ? correctURL : incorrectURL2;
+
+      let a1 = word.collection("intervention-set").doc("activity1").set({prompt: "Say " + wordValue, url: correctURL});
+      let a2 = word.collection("intervention-set").doc("activity2").set({correctUrl: correctURL, incorrectUrl: incorrectURL, prompt: "Click the image that is " + wordValue});
+      let a3 = word.collection("intervention-set").doc("activity3").set({correctAnswer: activity3Answer, prompt: "Is this image " + wordValue + "?", url: activity3URL});
+      let a4 = word.collection("intervention-set").doc("activity4").set({prompt: "Say " + wordValue, url: correctURL});
```

Comment:
This is really cool. You are creating different sets per person. I love this. If you could comment on how each intervention is being created that would be awesome!